### PR TITLE
Fix leaflet object

### LINKED ISSUE #49: can't change markers/icons with refs
Message from discord:

 ```
vue2-leaflet and it was like so: this.$refs.mapRef.$refs[markerRef][0].mapObject._icon.attributes.src.value = '/assets/img/icons/map_icon_orange.png' in vue-leaflet/vue-leaflet it seems that on map ref mapObject is replaced by leafletObject and marker ref or icon ref doesn't have neither mapObject nor leafletObject

Also in vue2-leaflet map ref would have more nested refs and I could get my icons from there, for example this.$refs.map.$refs.marker1, on vue-leaflet/vue-leaflet only nested ref in map is root, so only item: this.$refs.map.$refs.root which is not needed, better to have actual refs that are in  l-map slots

I found a way to change icons  with vue-leaflet/vue-leaflet by accessing map ref, then to leafletObject, then to _targets. Problem is that I have dynamic markers and target numbers are random, it's good when there's only one target on marker ref, but on map ref there will be tons of them.
```

So I'd see few issues
1. Ref on map doesn't expose object of refs used inside map as it did in vue2 (it does expose $refs{} but it only has root, no other refs)
2. MapObject is no longer exposed except for map, So it's now not possible to access `_icon` attribute or any other.

Plan from discord:
1. To expose leafletObject (renamed after 0.2.0 version from mapObject) on LMarker/LIcon

### PR DESCRIPTION
**Pull request by DonNicoJs**
Fixes #49 

This PR does the following:

- Brings out the render Fn from the `setup` otherwise we can't bind the `leafletObject` to the component instance
- Adds a `ready` event for every component that returns the `leafletObject`

This enables working with `refs` the only caveat: refs are really reliable to be present only in the `@ready` event handler OF each component so if I want to access a `leafletObject` of a marker i need to listen to the marker `@ready` this is because by async loading leaflet each component is initialised at different times.

### CODE REVIEW

**Comment by DonNicoJs** [comment]
@mikeu What do you think of this? It solves  a few problem in the codebase, but I had to virtually touch everything 😕

**Comment by DonNicoJs** [comment]
@saltibarsciai you may want to take a look here

**Comment by Saltibarsciai** [comment]
OMG, you did it! I'll checkout to fix-leaflet-object

**Comment by Saltibarsciai** [comment]
Tested on playground. works the same way as in vue2leaflet (except that mapObject is leafletObject). Thx. 
tested with:

`
 changeMyIcon(i){
      this.$refs['marker'+i].leafletObject._icon.attributes[0].nodeValue = 'https://images-na.ssl-images-amazon.com/images/I/51xy3obPFwL.png'
    }`

**Comment by mikeu** [comment]
Oh wow, thanks for this effort @DonNicoJs , what a PR! I'll have a chance to give it a proper review probably before you wake up Tuesday at the latest, I think, but certainly seems to be working well at first glance.

File: src/components/LGeoJson.vue
Comment (mikeu) [inline src/components/LGeoJson.vue:36]:
```suggestion
    return { ready, leafletObject: leafletRef };
```

File: src/components/LPolygon.vue
Comment (mikeu) [inline src/components/LPolygon.vue:41]:
```suggestion
    return { ready, leafletObject: leafletRef };
```

File: src/components/LPolyline.vue
Comment (mikeu) [inline src/components/LPolyline.vue:41]:
```suggestion
    return { ready, leafletObject: leafletRef };
```

File: src/components/LRectangle.vue
Comment (mikeu) [inline src/components/LRectangle.vue:45]:
```suggestion
    return { ready, leafletObject: leafletRef };
```

**Review by mikeu — APPROVED** [review]
Definitely happy for you to merge this after considering my minor suggestions, and move forward from here with this pattern!

File: src/components/LIcon.vue
Comment (DonNicoJs) [reply src/components/LIcon.vue:95]: The reason why I skipped l-icon completely is because is the only truly 'custom' component in my opinion (that does not bind 1:1 with a leaflet one). WDYT?

File: src/Playground.vue
Comment (DonNicoJs) [reply src/Playground.vue:170]: Left it on purpose as a way to import from the compiled bundle, but we can remove it for now

File: src/components/LCircle.vue
Comment (DonNicoJs) [reply src/components/LCircle.vue:44]: I agree! After this one is merged!

File: src/components/LControlAttribution.vue
Comment (DonNicoJs) [reply src/components/LControlAttribution.vue:25]: An oversight on my part! I will fix this!

File: src/components/LIcon.vue
Comment (mikeu) [reply src/components/LIcon.vue:95]: Yeah, I do agree that it is by far the most custom of the components, and `mapObject` was never exposed on it in Vue2Leaflet, so I agree it is certainly fine to leave `leafletObject` out here too!

There is still technically a Leaflet object inside the component that _could_ be exposed at some point, which is why I asked, although it would sometimes be `L.icon` and sometimes `L.divIcon`, and if nobody has asked for it in however many years of Vue2Leaflet anyway...

File: src/Playground.vue
Comment (mikeu) [reply src/Playground.vue]: No problem having it there if it's on purpose, just wanted to make sure :)

**Comment by DonNicoJs** [comment]
@mikeu I think I've addressed everything! Merging this!

**Pull request merged**
